documentation/documentation_compiler.ts

```diff
@@ -32,32 +33,62 @@ function compileSnippets() {
 function compileExamples() {
   const compiledExamples: CompiledExample[] = Examples.map(example => {
     const content = getContentFile(example.contentFile);
-    const code = compileCodeFiles(example);
+    const compiledExampleSnippets = compileExampleSnippets(example);
+    if (!isValidReferencePath(example.referencePath)) {
+      throw new Error(`${example.referencePath} is not a valid path`);
+    }
     return {
+      tokens: example.tokens,
+      referencePath: example.referencePath,
       content,
-      code,
-      categories: example.categories,
-      triggerTokens: example.triggerTokens,
+      exampleSnippets: compiledExampleSnippets,
     };
   });
+
   fs.writeFileSync(path.join(DocumentationRoot, 'generated/examples.json'), JSON.stringify(compiledExamples, null, 2));
 }
 
+function getCodeFile(file: string) {
+  const data = fs.readFileSync(path.join(DocumentationRoot, file), 'utf8');
+  const codeStart = data.indexOf(CodeBegin) + CodeBegin.length;
+  return data.substring(codeStart).trim();
+}
+
 function getContentFile(file: string) {
   return fs.readFileSync(path.join(DocumentationRoot, file), 'utf8').trim();
 }
 
-function compileCodeFiles(example: Example) {
-  const code = example.codeFiles.map(codeFile => {
-    return getCodeFile(codeFile);
+function compileExampleSnippets(example: Example): CompiledExampleSnippet[] {
+  return example.exampleSnippets.map(exampleSnippet => {
+    return {
+      name: formatSnippetName(path.basename(exampleSnippet.codeFile)),
+      content: exampleSnippet.content,
+      code: getCodeFile(exampleSnippet.codeFile),
+    };
   });
-  return code;
 }
 
-function getCodeFile(file: string) {
-  const data = fs.readFileSync(path.join(DocumentationRoot, file), 'utf8');
-  const codeStart = data.indexOf(codeBegin) + codeBegin.length;
-  return data.substring(codeStart).trim();
+function formatSnippetName(baseName: string): string {
+  const fileName = baseName.split('.')[0];
+  const snippetNameArray = fileName.split('_');
+  return snippetNameArray
+    .map(segment => {
+      return `${segment.charAt(0).toUpperCase()}${segment.slice(1)}`;
+    })
+    .join(' ');
+}
+
+function isValidReferencePath(referencePath: string): boolean {
```

Comment:
Can only handle reference paths that include a `fragmentIdentifier` (i.e. `.hmtl#fragmentIdentifier`). 

It's fine for the current use case of examples for builder function, but this can't validate paths to `types`